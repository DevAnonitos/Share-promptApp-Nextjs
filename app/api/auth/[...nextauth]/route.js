import NextAuth from "next-auth/next";
import GoogleProvider from "next-auth/providers/google";

const handler = NextAuth({
    providers: [
        GoogleProvider({
            clientId: process.env.GOOGLE_ID,
            clientSecret: process.env.GOOGLE_CLIENT_SECRET,
        })
    ],
    callbacks: {
        async session({ session }) {

        },
        async signIn({ account, profile, user, credentials }) {
            try {

            } catch (error) {
                console.log("Error checking if user exists: ", error.message);
                return false;
            }
        },
    }
});

export {
    handler as GET,
    handler as POST,
};
