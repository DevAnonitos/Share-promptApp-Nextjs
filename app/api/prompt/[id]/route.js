import Prompt from "../../../../models/prompt";
import { connectToDB } from "../../../../utils/database";

export const GET = async (request, { params }) => {
    try {
        await connectToDB();

        const prompt = await Prompt.findById(params.id).populate("creator");

        if(!prompt) {
            return new Response("Prompt not found", {
                status: 404,
            });
        }

        return new Response(JSON.stringify(prompt), {
            status: 200,
        });
    } catch (error) {
        return new Response("Internal Server Error", {
            status: 500,
        });
    }
};

export const PATH = async (request, { params }) => {
    const { prompt, tag } = await request.json();

    try {
        await connectToDB();

        const existingPrompt = await Prompt.findById(params.id);

        if(!existingPrompt) {
            return new Response("Prompt not founded", {
                status: 404,
            });
        }

        existingPrompt.prompt = prompt;
        existingPrompt.tag = tag;

        return new Response("Successfully update-prompt!", {
            status: 200,
        });

    } catch (error) {
        return new Response("Error updating Prompt", {
            status: 500,
        });
    }
};

export const DELETE = async (request, { params }) => {
    try {
        await connectToDB();

        await Prompt.findByIdAndRemove(params.id);

        return new Response("Prompt delete successfully", {
            status: 200,
        });
    } catch (error) {
        return new Response("Error deleting prompt", {
            status: 500,
        });
    }
};
