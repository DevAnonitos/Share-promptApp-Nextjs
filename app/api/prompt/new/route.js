import { connectToDB } from "@/utils/database";

export const POST = async (request) => {
    const { userId, prompt, tag } = await request.json();

    try {
        await connectToDB();

        return new Response(JSON.stringify(), {
            status: 201,
        })
    } catch (error) {
        return new Response("Fail to create a new prompt", {
            status: 500,
        })
    }
};
