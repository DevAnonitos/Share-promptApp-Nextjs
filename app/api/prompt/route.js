import Prompt from "@/models/prompt";
import { connectToDB } from "@/libs/utils/database";

export const GET = async (request) => {
    try {
        await connectToDB();

        const prompts = await Prompt.find({}).populate('creator');

        return new Response(JSON.stringify(prompts), {
            status: 200,
        });
    } catch (error) {
        return new Response(JSON("Fail to fetch All prompts", {
            status: 500,
        }))
    }
};
