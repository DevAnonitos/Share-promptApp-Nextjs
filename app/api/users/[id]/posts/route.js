import Prompt from "@/libs/models/prompt";
import { connectToDB } from "@/libs/utils/database";

export const GET = async (request, { params }) => {
    try {
        await connectToDB();

        if(!params || !params.id) {
            return new Response("UserId is missing", {
                status: 400,
            });
        }

        const prompts = await Prompt.find({ creator: params.id }).populate("creator");

        return new Response(JSON.stringify(prompts), {
            status: 200,
        });
    } catch (error) {
        console.log(error);
        return new Response("Failed to fetch prompts created by user", {
            status: 500
        });
    }
};
