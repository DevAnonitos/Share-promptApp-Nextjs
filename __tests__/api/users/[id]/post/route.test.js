/**
* @jest-environment node
*/
import { GET } from "@/app/api/users/[id]/posts/route";

describe("API User Prompt", () => {

  const baseUrl = "http://localhost:3000/api/users/64c52aa0183b5594893e6952/posts";

  it("should be return data json from /api/users/[id]/post", async () => {
    const res = await fetch(baseUrl);
    const data = await res.json();
    console.log(data);
    expect(data).toBeDefined();
    expect(data).toMatchSnapshot()
  })

  it("should be return status from /api/users/[id]/post", async () => {
    const res = await fetch(baseUrl);
    const data = await res.json();
    console.log(data);
    expect(res.status).toBe(200);
    expect(data).toMatchSnapshot()
  })
})