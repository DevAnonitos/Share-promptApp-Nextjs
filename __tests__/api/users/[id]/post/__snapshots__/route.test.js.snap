// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API User Prompt should be return data json from /api/users/[id]/post 1`] = `
[
  {
    "__v": 0,
    "_id": "65e9e4d144e66c1538cf36f1",
    "creator": {
      "__v": 0,
      "_id": "64c52aa0183b5594893e6952",
      "email": "nguyenbao2810dev@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AAcHTtd7ZyyZuG3rOtC70yT412sBj6dpowJIyJG3zqONzb58RQ=s96-c",
      "username": "nguyenbao2810dev",
    },
    "prompt": "testing App ne",
    "tag": "test",
  },
  {
    "__v": 0,
    "_id": "65f50633ac1867d4610726cc",
    "creator": {
      "__v": 0,
      "_id": "64c52aa0183b5594893e6952",
      "email": "nguyenbao2810dev@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AAcHTtd7ZyyZuG3rOtC70yT412sBj6dpowJIyJG3zqONzb58RQ=s96-c",
      "username": "nguyenbao2810dev",
    },
    "prompt": "HDR photo of woman with hair in a messy bun smiles at the viewer, perfect eyes, beautiful skin, outside on a warm sunny day wearing a summer dress . High dynamic range, vivid, rich details, clear shadows and highlights, realistic, intense, enhanced contrast, highly detailed",
    "tag": "stable diffusion",
  },
  {
    "__v": 0,
    "_id": "65f506c8ac1867d4610726d4",
    "creator": {
      "__v": 0,
      "_id": "64c52aa0183b5594893e6952",
      "email": "nguyenbao2810dev@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AAcHTtd7ZyyZuG3rOtC70yT412sBj6dpowJIyJG3zqONzb58RQ=s96-c",
      "username": "nguyenbao2810dev",
    },
    "prompt": "A portrait of a young woman with sun-kissed blonde wavy hair and light hazel eyes. She has a clear complexion with a scattering of freckles across her nose and cheeks. Her expression is contemplative as she gazes towards the viewer. She wears minimal makeup, with a focus on her eyes, and an elegant white or silver drop earring. She is in a simple black sleeveless top, against a soft, natural background suggesting an outdoor setting. The lighting is soft and natural, highlighting her features with a serene ambiance.
  
",
    "tag": "stable diffusion",
  },
]
`;

exports[`API User Prompt should be return status from /api/users/[id]/post 1`] = `
[
  {
    "__v": 0,
    "_id": "65e9e4d144e66c1538cf36f1",
    "creator": {
      "__v": 0,
      "_id": "64c52aa0183b5594893e6952",
      "email": "nguyenbao2810dev@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AAcHTtd7ZyyZuG3rOtC70yT412sBj6dpowJIyJG3zqONzb58RQ=s96-c",
      "username": "nguyenbao2810dev",
    },
    "prompt": "testing App ne",
    "tag": "test",
  },
  {
    "__v": 0,
    "_id": "65f50633ac1867d4610726cc",
    "creator": {
      "__v": 0,
      "_id": "64c52aa0183b5594893e6952",
      "email": "nguyenbao2810dev@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AAcHTtd7ZyyZuG3rOtC70yT412sBj6dpowJIyJG3zqONzb58RQ=s96-c",
      "username": "nguyenbao2810dev",
    },
    "prompt": "HDR photo of woman with hair in a messy bun smiles at the viewer, perfect eyes, beautiful skin, outside on a warm sunny day wearing a summer dress . High dynamic range, vivid, rich details, clear shadows and highlights, realistic, intense, enhanced contrast, highly detailed",
    "tag": "stable diffusion",
  },
  {
    "__v": 0,
    "_id": "65f506c8ac1867d4610726d4",
    "creator": {
      "__v": 0,
      "_id": "64c52aa0183b5594893e6952",
      "email": "nguyenbao2810dev@gmail.com",
      "image": "https://lh3.googleusercontent.com/a/AAcHTtd7ZyyZuG3rOtC70yT412sBj6dpowJIyJG3zqONzb58RQ=s96-c",
      "username": "nguyenbao2810dev",
    },
    "prompt": "A portrait of a young woman with sun-kissed blonde wavy hair and light hazel eyes. She has a clear complexion with a scattering of freckles across her nose and cheeks. Her expression is contemplative as she gazes towards the viewer. She wears minimal makeup, with a focus on her eyes, and an elegant white or silver drop earring. She is in a simple black sleeveless top, against a soft, natural background suggesting an outdoor setting. The lighting is soft and natural, highlighting her features with a serene ambiance.
  
",
    "tag": "stable diffusion",
  },
]
`;
